### PR TITLE
Upgrade project to Python 3.11

### PR DESCRIPTION
**Pull request by codders**
Upgrade to the latest python.
Removed the (unmaintaned) acora search library and replaced it with Intel hyperscan
Rearranged the test suite and added some performance improvements to reduce total test time.

### CODE REVIEW

File: w4af/core/data/quick_match/multi_re.py [hunk 1]
```diff
@@ -70,96 +95,103 @@ def _build(self):
...
+    def query(self, target_str: bytes) -> Generator[Tuple[Match, str, bytes, Optional[Any]], None, None]:
         """
         Run through all the regular expressions and identify them in target_str.
 
         We'll only run the regular expressions if:
              * They do not have keywords
              * The keywords exist in the string
 
-        :param target_str: The target string where the keywords need to be match
+        :param target_str: The target string where the keywors need to be match
```
Comment (security-companion) [inline w4af/core/data/quick_match/multi_re.py:130]: a d is missing in keywords

File: Pipfile [hunk 2]
```diff
@@ -1,92 +1,79 @@
...
-acora = "*"
+pebble = "*"
 diff-match-patch = "*"
 bravado-core = "*"
 lz4 = "*"
 vulners = "*"
 ipaddresses = "*"
 xdot = "*"
-nocasedict = "*"
+nocasedict = "==1.0.4"
```
Comment (security-companion) [inline Pipfile:39]: Could you explain why this specific version is used?

File: Pipfile [hunk 3]
```diff
@@ -1,92 +1,79 @@
 [[source]]
-url = "https://pypi.python.org/simple"
+url = "https://pypi.org/simple"
```
Comment (security-companion) [inline Pipfile:2]: Question: Did you use a specific command/tool for rearraning the Pipfile and pipfile.lock or do this manually?

File: w4af/plugins/attack/db/sqlmap/LICENSE [hunk 4]
```diff
@@ -1,7 +1,7 @@
 COPYING -- Describes the terms under which sqlmap is distributed. A copy
 of the GNU General Public License (GPL) is appended to this file.
 
-sqlmap is (C) 2006-2022 Bernardo Damele Assumpcao Guimaraes, Miroslav Stampar.
+sqlmap is (C) 2006-2023 Bernardo Damele Assumpcao Guimaraes, Miroslav Stampar.
```
Comment (security-companion) [inline w4af/plugins/attack/db/sqlmap/LICENSE:4]: Do I assume right that you updated sqlmap? Which is the used version/git commit hash of sqlmap?

**Review by security-companion — APPROVED** [review]
Thank you for your work,
I added some comments
What I can do (perhaps in a separate commit) is update the documentation (eg. \w4af\doc\sphinx\install.rst still contains python 3.10)

File: w4af/core/data/quick_match/multi_re.py (hunk 1 above)
Comment (codders) [reply w4af/core/data/quick_match/multi_re.py:130]: well spotted!

File: Pipfile (hunk 2 above)
Comment (codders) [reply Pipfile:39]: Yup. I made a comment in the commit, but there is currently a bug / incompatibility with newer versions: https://github.com/pywbem/nocasedict/issues/139

File: Pipfile (hunk 3 above)
Comment (codders) [reply Pipfile:2]: I had so many incompatibilities with the upgrade, that I ended up deleting the pip and lock files and starting over. So this was generated by the latest version of Pipenv

File: w4af/plugins/attack/db/sqlmap/LICENSE (hunk 4 above)
Comment (codders) [reply w4af/plugins/attack/db/sqlmap/LICENSE:4]: Yup. There's a unit test we have that checks that the sqlmap version is not out of date - after 90 days we need to hit the update button. According to the merge commit (our `2a56436cea0e85eb9a4af56a1f6597aeccf3131b`) this was commit `c5d7c542a` of sqlmap (just taken from the master branch).

**Pull request merged**